containers/src/list.c

```diff
@@ -117,10 +118,19 @@ void * list_at(list * list, unsigned int position)
 	return finder->_data;
 }
 
+/**
+ * list_assign - assigns right list to the left list
+ *
+ * @left - left list
+ * @right - right list
+ *
+ * May be called with valid arguments.
+ */
 int list_assign(list * left, list * right)
 {
 	if (right == NULL) {
-		return LIST_ASSIGN_RIGHT_IS_NULL;
+		left = NULL;
+		return 0;
 	}
 	else if (list_is_empty(right)) {
 		left->_top = left->_last = NULL;
```

Comment:
Note **list_destroy** can't be called in this execution flow, becuase you return 0.

``` list* left = list_create(...);
    list* right = list_create(...);
    int i = 0;
    while(i++ < 4000) {
        some_big_structure* some_big_data = malloc(sizeof(some_big_structure));
        list_push_back(left, right);
    }
    list_assign(letf, right);
  
     //  memory of 4000 elements are not released, addresses are corrupted because you wrote this:
     //  left->top = left->last = NULL; in assign(list*,list*)
     //  Instead you should clear content of left list before assigning NULL.
     //  Like this : 
     //  list_clear(left);
     //  left->top = left->last = NULL;